src/overlays/actors/ovl_En_Baguo/z_en_baguo.c

```diff
@@ -27,92 +38,458 @@ const ActorInit En_Baguo_InitVars = {
-#endif
+extern Gfx D_060014C8;
+extern Gfx D_060018C8;
+extern Gfx D_06001CC8;
+extern SkeletonHeader D_060020E8;
+
+void EnBaguo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnBaguo* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 0.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_060020E8, NULL, this->jointTable, this->morphTable, 3);
+    this->actor.hintId = 0xB;
+    this->maxDistanceFromHome = 240.0f;
+    this->maxDistanceFromHome += this->actor.world.rot.z * 40.0f;
+    this->actor.world.rot.z = 0;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.targetMode = 2;
+    Collider_InitAndSetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->colliderElements);
+    this->collider.elements[0].dim.modelSphere.radius = 30;
+    this->collider.elements[0].dim.scale = 1.0f;
+    this->collider.elements[0].dim.modelSphere.center.x = 80;
+    this->collider.elements[0].dim.modelSphere.center.y = 80;
+    this->collider.elements[0].dim.modelSphere.center.z = 0;
+    this->actor.shape.yOffset = -3000.0f;
+    this->actor.gravity = -3.0f;
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+    this->actor.flags |= 0x8000000;
+    this->actor.flags &= ~1;
+    this->collider.base.acFlags |= AC_HARD;
+    this->actionFunc = EnBaguo_UndergroundIdle;
+}
+
+void EnBaguo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBaguo* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void EnBaguo_UndergroundIdle(EnBaguo* this, GlobalContext* globalCtx) {
+    this->action = NEJIRON_ACTION_INACTIVE;
+    if (this->actor.xzDistToPlayer < 200.0f && Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK) {
+        this->actor.draw = EnBaguo_DrawBody;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_BAKUO_APPEAR);
+        this->actor.world.rot.z = 0;
+        this->actor.world.rot.x = this->actor.world.rot.z;
+        this->actor.flags &= ~0x8000000;
+        this->actor.flags |= 1;
+        this->actionFunc = EnBaguo_EmergeFromUnderground;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+void EnBaguo_EmergeFromUnderground(EnBaguo* this, GlobalContext* globalCtx) {
+    this->actor.world.rot.y += 0x1518;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    if ((globalCtx->gameplayFrames % 8) == 0) {
+        func_800BBDAC(globalCtx, &this->actor, &this->actor.world.pos, this->actor.shape.shadowScale - 20.0f, 10, 8.0f,
+                      500, 10, 1);
+    }
+    Math_ApproachF(&this->actor.shape.shadowScale, 50.0f, 0.3f, 5.0f);
+    Math_ApproachF(&this->actor.shape.yOffset, 2700.0f, 100.0f, 500.0f);
+    if (this->actor.shape.yOffset > 2650.0f) {
+        this->action = NEJIRON_ACTION_ACTIVE;
+        this->actor.shape.yOffset = 2700.0f;
+        this->timer = 60;
+        this->actionFunc = EnBaguo_Sit;
+    }
+}
+
+void EnBaguo_Sit(EnBaguo* this, GlobalContext* globalCtx) {
```

Comment:
I think this may be more of an idle since it's the state the nejiron is in while it waits to roll. ReadyRoll may also be a good name for it.